test/spectests/spectests.cpp

```diff
@@ -268,17 +291,30 @@ void run_tests_from_dir(const fs::path& path)
 
     std::sort(std::begin(files), std::end(files));
 
+    test_results total;
+    bool exception_thrown = false;
     for (const auto& f : files)
     {
         try
         {
-            test_runner{}.run_from_file(f);
+            const auto res = test_runner{settings}.run_from_file(f);
```

Comment:
Since the internal `test_results` are not reset in `run_from_file`, one could just instantiate the runner once and collect the results once.

Reply:
There's also `test_runner::instance` and at some point it might become a number of instances for test files that link several modules together. Therefore I'd like to leave `test_runner` a class that handles running tests from one file.